tests/integration/test_ethernet.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import pytest
+import subprocess
+import sys
+
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+
+parentdir = os.path.normpath(os.path.join(os.path.dirname(__file__), "../../"))
+with mock.patch.object(
+    sys,
+    "path",
+    [parentdir, os.path.join(parentdir, "module_utils/network_lsr")] + sys.path,
+):
+    with mock.patch.dict(
+        "sys.modules",
+        {
+            "ansible": mock.Mock(),
+            "ansible.module_utils": __import__("module_utils"),
+            "ansible.module_utils.basic": mock.Mock(),
+        },
+    ):
+        import library.network_connections as nc
+
+
+class PytestRunEnvironment(nc.RunEnvironment):
+    def log(self, connections, idx, severity, msg, **kwargs):
+        if severity == nc.LogLevel.ERROR:
+            print("Error: {}".format(connections[idx]))
+            raise RuntimeError(msg)
+        else:
+            print("Log: {}".format(connections[idx]))
+
+    def run_command(self, argv, encoding=None):
+        command = subprocess.Popen(argv)
+        return_code = command.wait()
+        out, err = command.communicate()
+        return return_code, out, err
+
+    def _check_mode_changed(self, *args, **kwargs):
+        pass
+
+
+def _configure_network(connections, provider):
+    cmd = nc.Cmd.create(
+        provider,
+        run_env=PytestRunEnvironment(),
+        connections_unvalidated=connections,
+        connection_validator=nc.ArgValidator_ListConnections(),
+    )
+    cmd.run()
+
+
+@pytest.fixture(scope="session")
+def provider(request):
+    return request.config.getoption("--provider")
+
+
+@pytest.fixture
+def testnic1():
+    veth_name = "testeth"
+    try:
+        create_iface = subprocess.Popen(
+            [
+                "ip",
+                "link",
+                "add",
+                veth_name,
+                "type",
+                "veth",
+                "peer",
+                "name",
+                veth_name + "peer",
+            ],
+            close_fds=True,
+        )
+        create_iface.wait()
+        yield veth_name
+    finally:
+        delete_iface = subprocess.Popen(["ip", "link", "delete", veth_name])
```

Comment:
If you do not need stdout, you can use `subprocess.call()` to simplify things:
https://docs.python.org/2.7/library/subprocess.html#using-the-subprocess-module

Reply:
I see, thanks!